src/objc-renderer.ts

```diff
@@ -243,14 +243,77 @@ function enumerationIsPublic(isPublic:boolean):(enumeration:ObjC.Enumeration) =>
   };
 }
 
+const NS_ASSUME_NONNULL_BEGIN:string = 'NS_ASSUME_NONNULL_BEGIN';
+const NS_ASSUME_NONNULL_END:string = 'NS_ASSUME_NONNULL_END';
+
+function prefixClassMacros(classInfo:ObjC.Class):string[] {
+  var macros:string[] = [];
+  switch (classInfo.nullability) {
+    case ObjC.ClassNullability.assumeNonnull:
+      macros.push(NS_ASSUME_NONNULL_BEGIN);
+      break;
+    case ObjC.ClassNullability.default:
+      break;
+  }
+  return macros;
+}
+
+function postfixClassMacros(classInfo:ObjC.Class):string[] {
+  var macros:string[] = [];
+  switch (classInfo.nullability) {
+    case ObjC.ClassNullability.assumeNonnull:
+      macros.push(NS_ASSUME_NONNULL_END);
+      break;
+    case ObjC.ClassNullability.default:
+      break;
+  }
+  return macros;
+}
+
+function prefixBlockTypeMacros(blockType:ObjC.BlockType):string[] {
+  var macros:string[] = [];
+  switch (blockType.nullability) {
+    case ObjC.ClassNullability.assumeNonnull:
+      macros.push(NS_ASSUME_NONNULL_BEGIN);
+      break;
+    case ObjC.ClassNullability.default:
+      break;
+  }
+  return macros;
+}
+
+function postfixBlockTypeMacros(blockType:ObjC.BlockType):string[] {
+  var macros:string[] = [];
+  switch (blockType.nullability) {
+    case ObjC.ClassNullability.assumeNonnull:
+      macros.push(NS_ASSUME_NONNULL_END);
```

Comment:
I kinda thought about that, but then saw that most of everything else was structured this way. But yeah refactoring this into a `Macro` type and a `MacroLocation` type would make this cleaner. I'll refactor them.